code/go/internal/validator/semantic/types.go

```diff
@@ -33,41 +34,50 @@ type field struct {
 	Fields fields `yaml:"fields"`
 }
 
-type validateFunc func(fieldsFile string, f field) ve.ValidationErrors
+type fieldFileMetadata struct {
+	packageType  string
+	packageName  string
+	dataStream   string
+	filePath     string
+	fullFilePath string
+}
+
+type validateFunc func(fileMetadata fieldFileMetadata, f field) ve.ValidationErrors
 
 func validateFields(fsys fspath.FS, validate validateFunc) ve.ValidationErrors {
-	fieldsFiles, err := listFieldsFiles(fsys)
+
+	fieldsFilesMetadata, err := listFieldsFiles(fsys)
 	if err != nil {
 		return ve.ValidationErrors{errors.Wrap(err, "can't list fields files")}
 	}
 
 	var vErrs ve.ValidationErrors
-	for _, fieldsFile := range fieldsFiles {
-		unmarshaled, err := unmarshalFields(fsys, fieldsFile)
+	for _, metadata := range fieldsFilesMetadata {
+		unmarshaled, err := unmarshalFields(fsys, metadata.filePath)
 		if err != nil {
-			vErrs = append(vErrs, errors.Wrapf(err, `file "%s" is invalid: can't unmarshal fields`, fsys.Path(fieldsFile)))
+			vErrs = append(vErrs, errors.Wrapf(err, `file "%s" is invalid: can't unmarshal fields`, fsys.Path(metadata.filePath)))
```

Comment:
Could we use `metadata.fullFilePath` here?
```suggestion
			vErrs = append(vErrs, errors.Wrapf(err, `file "%s" is invalid: can't unmarshal fields`, metadata.fullFilePath))
```

Reply:
At first, I wanted to ensure that the same errors were provided.
Now, that it has been checked that, I think this could be updated as you suggest.

Reply:
Doing some tests, `metadata.filePath` is required to unmarshal the fields. It requires the path inside the package to the fields file.

```go
		unmarshaled, err := unmarshalFields(fsys, metadata.filePath)
		if err != nil {
			vErrs = append(vErrs, errors.Wrapf(err, `file "%s" is invalid: can't unmarshal fields`, metadata.fullFilePath))
		}
```